File: Examples/src/main/java/com/aspose/words/examples/programming_documents/styles/ExtractContentBasedOnStyles.java
package com.aspose.words.examples.programming_documents.styles;

import com.aspose.words.*;
import com.aspose.words.examples.Utils;
import com.aspose.words.examples.programming_documents.tables.ApplyFormatting.ApplyBordersAndShading;
import java.util.ArrayList;

/**
 * Shows how to find paragraphs and runs formatted with a specific style.
 */
public class ExtractContentBasedOnStyles {

	private static final String dataDir = Utils.getSharedDataDir(ApplyBordersAndShading.class) + "Styles/";

	public static void main(String[] args) throws Exception {
		// Open the document.
		Document doc = new Document(dataDir + "TestFile.doc");

		// Define style names as they are specified in the Word document.
		final String PARA_STYLE = "Heading 1";
		final String RUN_STYLE = "Intense Emphasis";

		// Collect paragraphs with defined styles.
		// Show the number of collected paragraphs and display the text of this paragraphs.
		ArrayList<Paragraph> paragraphs = paragraphsByStyleName(doc, PARA_STYLE);
		System.out.println(java.text.MessageFormat.format("Paragraphs with \"{0}\" styles ({1}):", PARA_STYLE, paragraphs.size()));
		for (Paragraph paragraph : paragraphs)
			System.out.print(paragraph.toString(SaveFormat.TEXT));

		// Collect runs with defined styles.
		// Show the number of collected runs and display the text of this runs.
		ArrayList<Run> runs = runsByStyleName(doc, RUN_STYLE);
		System.out.println(java.text.MessageFormat.format("\nRuns with \"{0}\" styles ({1}):", RUN_STYLE, runs.size()));
		for (Run run : runs)
			System.out.println(run.getRange().getText());
	}

	public static ArrayList<Paragraph> paragraphsByStyleName(Document doc, String styleName) throws Exception {
		// Create an array to collect paragraphs of the specified style.
		ArrayList<Paragraph> paragraphsWithStyle = new ArrayList();
		// Get all paragraphs from the document.
		NodeCollection paragraphs = doc.getChildNodes(NodeType.PARAGRAPH, true);
		// Look through all paragraphs to find those with the specified style.
		for (Paragraph paragraph : (Iterable<Paragraph>) paragraphs) {
			if (paragraph.getParagraphFormat().getStyle().getName().equals(styleName))
				paragraphsWithStyle.add(paragraph);
		}
		return paragraphsWithStyle;
	}

	public static ArrayList<Run> runsByStyleName(Document doc, String styleName) throws Exception {
		// Create an array to collect runs of the specified style.
		ArrayList<Run> runsWithStyle = new ArrayList();
		// Get all runs from the document.
		NodeCollection runs = doc.getChildNodes(NodeType.RUN, true);
		// Look through all runs to find those with the specified style.
		for (Run run : (Iterable<Run>) runs) {
			if (run.getFont().getStyle().getName().equals(styleName))
				runsWithStyle.add(run);
		}
		return runsWithStyle;
	}
}

File: Plugins/Aspose_Words_Java_for_Jython/asposewords/programming_documents/ExtractContentBasedOnStyles.py
from asposewords import Settings
from com.aspose.words import Document
from com.aspose.words import SaveFormat
from com.aspose.words import NodeType

class ExtractContentBasedOnStyles:

    def __init__(self):
        dataDir = Settings.dataDir + 'programming_documents/'
        
        # Open the document.
        doc = Document(dataDir + "TestFile.doc")

        # Define style names as they are specified in the Word document.
        PARA_STYLE = "Heading 1"
        RUN_STYLE = "Intense Emphasis"

        # Collect paragraphs with defined styles.
        # Show the number of collected paragraphs and display the text of this paragraphs.
        paragraphs = self.paragraphs_by_style_name(doc, PARA_STYLE)

        print "abc = " + str(paragraphs[0])
        print "Paragraphs with " + PARA_STYLE + " styles " + str(len(paragraphs)) + ":"

        for paragraph in paragraphs :
            print str(paragraph.toString(SaveFormat.TEXT))

        # Collect runs with defined styles.
        # Show the number of collected runs and display the text of this runs.
        runs = self.runs_by_style_name(doc, RUN_STYLE)

        print "Runs with " + RUN_STYLE + " styles " + str(len(runs)) + ":"

        for run in runs :
            print run.getRange().getText()
    
    def paragraphs_by_style_name(self, doc, styleName):

        # Create an array to collect paragraphs of the specified style.
        paragraphsWithStyle = []
        # Get all paragraphs from the document.
        paragraphs = doc.getChildNodes(NodeType.PARAGRAPH, True)
        # Look through all paragraphs to find those with the specified style.

        paragraphs_count = paragraphs.getCount()

        i = 0
        while(i < paragraphs_count) :
            paragraph = paragraphs.get(i)
            if (paragraph.getParagraphFormat().getStyle().getName() == styleName):
                paragraphsWithStyle.append(paragraph)
            i = i + 1

        return paragraphsWithStyle

    def runs_by_style_name(self, doc, styleName):

        # Create an array to collect runs of the specified style.
        runsWithStyle = []

        runs = doc.getChildNodes(NodeType.RUN, True)
        # Look through all runs to find those with the specified style.
        runs = runs.toArray()
        for run in runs :
            if (run.getFont().getStyle().getName() == styleName):
                runsWithStyle.append(run)

        return runsWithStyle

if __name__ == '__main__':        
    ExtractContentBasedOnStyles()